### PR TITLE
Added defs for undocumented attrs on enums

### LINKED ISSUE #2934: Type definition for Enum is missing some attributes
The [attribute definition](https://github.com/python/typeshed/blob/master/stdlib/3/enum.pyi#L23-L27) skips a few attributes. For example [`_member_map_`](https://github.com/python/cpython/blob/8b914d2767acba3a9e78f1dacdc2d61dbfd7e304/Lib/enum.py#L166) and [`_value2member_map_`](https://github.com/python/cpython/blob/8b914d2767acba3a9e78f1dacdc2d61dbfd7e304/Lib/enum.py#L176).

### PR DESCRIPTION
**Pull request by jetpacktuxedo**
This should resolve #2934

I think I followed the contributing steps correctly, but I'm more than happy to tweak things as needed!

### CODE REVIEW

File: stdlib/3/enum.pyi [hunk 1]
```diff
@@ -25,6 +25,9 @@ class Enum(metaclass=EnumMeta):
     value: Any
     _name_: str
     _value_: Any
+    _member_names: List[str]  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/3/enum.pyi]: `_member_names_` actually.

File: stdlib/3/enum.pyi [hunk 2]
```diff
@@ -25,6 +25,9 @@ class Enum(metaclass=EnumMeta):
...
+    _member_map_: Dict[str, object]  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/3/enum.pyi]: The value type for these two can be `Enum` I think.

File: stdlib/3/enum.pyi [hunk 3]
```diff
@@ -25,6 +25,9 @@ class Enum(metaclass=EnumMeta):
...
+    _value2member_map: Dict[int, object]  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/3/enum.pyi]: `_value2member_map_`

**Comment by jetpacktuxedo** [comment]
Thanks for catching those! Just dropped back in and touched them up and _hopefully_ fixed that ci failure the right way.

**Pull request merged**